src/client/browser/index.js

```diff
@@ -9,9 +9,9 @@ const HEARTBEAT_INTERVAL = 30 * 1000;
 //Utils
 // NOTE: the window.XMLHttpRequest may have been wrapped by Hammerhead, while we should send a request to
 // the original URL. That's why we need the XMLHttpRequest argument to send the request via native methods.
-function sendXHR (url, XMLHttpRequest) {
+function sendXHR (url, createXHR) {
     return new Promise((resolve, reject) => {
-        var xhr = new XMLHttpRequest();
+        var xhr = new createXHR();
 
```

Comment:
[The comment](https://github.com/DevExpress/testcafe/pull/473/files#r60551562) by @AlexanderMoskovkin was about that on test page, under `hammerhead`, `createXHR` is not a constructor, just a plain function, returning value. So it's not nice use it that way, despite it works (see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new), why).

So it's must be:
```js
var xhr = createXHR()
```

But also consider that on `Idle Page`, where is no `hammerhead`, a constructor function will be passed as `createXHR` (check [here](https://github.com/helen-dikareva/testcafe-phoenix/blob/41fa162110602dd1a254bac4bebbf28371cfe50c/src/client/browser/idle-page/index.js#L14)).  
 
So I suppose you need to create your own `createNativeXHR` for the idle page, and use it there.